### ISSUE TITLE
Serialization Failure in Clockwork Due to Incomplete Object Representations

### ISSUE THREAD

**Issue by hamedghaderi** [itsgoingd/clockwork#695]
We have encountered an issue with Clockwork where serialization fails under certain conditions. This problem appears to stem from incomplete object representations during operations that require serialization.

**Comment by UlrichEckhardt** [comment]
Looking at the PR, it seems to be a harmless change. Thinking about maintainability though, I'm wondering how you came across that error to begin with. What even is `__PHP_Incomplete_Class`? The info I found was almost none. Can you perhaps describe how to reproduce this?

**Comment by hamedghaderi** [comment]
@UlrichEckhardt, the occurrence of the `__PHP_Incomplete_Class` error in our application stems from our multi-step model construction process. In this process, we incrementally validate and assemble different segments of a PHP class, with each validation step contributing to the formation of the model. To facilitate user progress through these steps, we temporarily store the partially completed model in the session.

This approach allows us to preserve and reuse the already validated segments of the model, ensuring that the user does not need to re-enter data upon encountering a validation error. It is only upon reaching the final step that we compile the fully qualified PHP class and persist it to the database.

**Comment by hamedghaderi** [comment]
@UlrichEckhardt To add clarity to our approach, let's consider a simplified example similar to our implementation. In our application, to persist user progress and validated data between these steps, we employ session storage:

```php
function saveModelToSession($model) {
    $sessionData = $this->session->get('model_data');
    $modelClass = get_class($model);

    if (!isset($sessionData[$modelClass])) {
        $sessionData[$modelClass] = [];
    }

    $sessionData[$modelClass][$model->id] = $model;
   $this->session->set('model_data', $sessionData);
}
```

**Comment by itsgoingd** [comment]
Hey, thanks for investigating this and making a PR.

To reproduce this you can simply call `unserialize()` with a non-existent serialized class:

```php
unserialize('O:3:"Foo":0:{}'); // __PHP_Incomplete_Class(Foo)
```

**Comment by itsgoingd** [comment]
Released Clockwork 5.2.1, including this fix.
